app.js

```diff
@@ -1,141 +1,214 @@
+        data.sessions[i].min_age_limit > 18
+      ) {
+        if (
+          document.getElementById('optionsDose1').checked &&
+          document.getElementById('optionsDose2').checked
+        ) {
+          //show both dose 1 & dose 2 details
+          //Note: Here dose 1 or dose 2 avaialability is note checked, because if neither of them was available, then this code block would not run
+          //This code block is sub clause of top level If statement that runs only if (available_dose > 0)
+          insertRowInTable(
+            districtID,
+            districtName,
+            `45+ Any Dose`,
+            data.sessions[i].name,
+            data.sessions[i].pincode,
+            data.sessions[i].available_capacity_dose1,
+            data.sessions[i].available_capacity_dose2
+          )
+        } else if (
+          document.getElementById('optionsDose1').checked &&
+          data.sessions[i].available_capacity_dose1 > 0
+        ) {
+          //show dose 1 details only
+          insertRowInTable(
+            districtID,
+            districtName,
+            `45+ Dose 1`,
+            data.sessions[i].name,
+            data.sessions[i].pincode,
+            data.sessions[i].available_capacity_dose1,
+            'NA'
+          )
+        } else if (
+          document.getElementById('optionsDose2').checked &&
+          data.sessions[i].available_capacity_dose2 > 0
+        ) {
+          //show dose 2 details only
+          insertRowInTable(
+            districtID,
+            districtName,
+            `45+ Dose 2`,
+            data.sessions[i].name,
+            data.sessions[i].pincode,
+            'NA',
+            data.sessions[i].available_capacity_dose2
+          )
         }
+      }
     }
-
+  }
 }
 function search() {
-    //populating updated values from user input form into dateOfAvailability and inputDistrictCodes;
-    dateOfAvailability = document.getElementById('dateForCheckingAvailability').value;
-    inputDistrictCodes = document.getElementById('disctrictCodes').value;
-    console.log(dateOfAvailability);
-    console.log(inputDistrictCodes);
-    inputDistrictCodes = inputDistrictCodes.split(",");
-    document.getElementById('refreshTimerText').innerText = `Auto Refreshes every ${inputDistrictCodes.length * 3} seconds`;
-    //clear existing records
-    if (table.rows.length > 1) {
-        console.log(table.rows.length);
-        for (i = -1 * table.rows.length; i < -1; i++) {
-            table.deleteRow(i * -1 - 1);
-        }
+  //populating updated values from user input form into dateOfAvailability and inputDistrictCodes;
+  dateOfAvailability = document.getElementById('dateForCheckingAvailability').value
+  inputDistrictCodes = document.getElementById('disctrictCodes').value
+  console.log(dateOfAvailability)
+  console.log(inputDistrictCodes)
+  inputDistrictCodes = inputDistrictCodes.split(',')
+  document.getElementById('refreshTimerText').innerText = `Auto Refreshes every ${
```

Comment:
Lets avoid the whitespace in the Template String .
This is more readable
```
Auto Refreshes every ${inputDistrictCodes.length * 3} seconds`;
```

In comparison to this:
```
document.getElementById('refreshTimerText').innerText = `Auto Refreshes every ${
   inputDistrictCodes.length * 3
  } seconds`
```

Reply:
Suggest we just remove app.js from the PR entirely. I initially was going to make changes, but decided to refactor into a separate file.